### PR TITLE
docs: Add Provisioning saga workflow ADR

### PR DESCRIPTION
**Pull request by adiabramovitch**
Adding an ADR explaining the three option I have come up with regarding the Provisioning saga workflow

### CODE REVIEW

File: docs/adr/013-add-provisioning-saga.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+## Problem Statement
+
+This ADR exists because there is a need of making the build and launch workflow more user-friendly. 
+this ADR will include three propositions regarding how should we implement this feature and what should we take into consideration.
+
+## Goals
+
+* Make the saga workflow fast, secure, and simple
+* Choose one of the architectures prop3os3ed
```
Comment (ezr-ondrej) [inline docs/adr/013-add-provisioning-saga.md:17]: encoding error? :thinking: xD

I'd maybe go with `Decide on the solution architecture`

File: docs/adr/013-add-provisioning-saga.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Important information to remember:
+
+* The Provisioning is dependent on the image id in order to continue and finish the launch
+* The shared sources are given directly from the image variable from IB Wizard
+
+**In the following section I will elaborate on the differences:**
+
+1. Combining Image Builder wizard with the Provisioning wizard and showing the status of the instance on Image Builder main page:
+   * Add a pooling mechanism questioning Image Builder if the image is ready.
```
Comment (ezr-ondrej) [inline docs/adr/013-add-provisioning-saga.md:45]:
```suggestion
   * Add a polling to Image Builder for image readieness.
```

File: docs/adr/013-add-provisioning-saga.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+* The Provisioning wizard opens up
+
+Important information to remember:
+
+* The Provisioning is dependent on the image id in order to continue and finish the launch
+* The shared sources are given directly from the image variable from IB Wizard
+
+**In the following section I will elaborate on the differences:**
+
+1. Combining Image Builder wizard with the Provisioning wizard and showing the status of the instance on Image Builder main page:
```
Comment (ezr-ondrej) [inline docs/adr/013-add-provisioning-saga.md:44]: I'm not clear what's first, when do we poll and where (the Prov UI, prov backend - background job, IB?)

File: docs/adr/013-add-provisioning-saga.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+   * Add a pooling mechanism questioning Image Builder if the image is ready.
+   * The customer fills in the data needed in order to launch an image.
+   * The wizard closes, the pooling continues, and when it is done a job is enqueued with the image id.
+   * The status of the job is presented on Image Builder main page. (like the status is being presented in the wizard now)
+  
+2. Combining Image Builder wizard with the provisioning wizard and adding a step in the Provisioning wizard creating the image:
+   * The customer fills in the data needed in order to launch an image.
+   * We are adding a step in the Provisioning wizard showing the status of the image creation.
```
Comment (ezr-ondrej) [inline docs/adr/013-add-provisioning-saga.md:52]:
```suggestion
   * There is an additional step before all the others in the Provisioning wizard showing the status of the image build.
```

File: docs/adr/013-add-provisioning-saga.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+   * After the image has been created the Provisioning workflow continues(creating a reservation, uploading pubkeys when needed and launching the image)
+  
+3. Combining Image Builder wizard with the provisioning wizard and showing the status of the instance on Inventory main page:
+   * Add a pooling mechanism questioning Image Builder if the image is ready.
```
Comment (ezr-ondrej) [inline docs/adr/013-add-provisioning-saga.md:56]: ditto

**Review by ezr-ondrej — COMMENTED** [review]
Just few wording nits, otherwise I guess we are good here, very nice write up.
I'm all up solution 3. but lets merge it proposed here for now and decide once we discuss it with Inventory and Image builder :+1:

File: docs/adr/013-add-provisioning-saga.md (hunk 5 above)
Comment (adiabramovitch) [reply docs/adr/013-add-provisioning-saga.md:56]: ditto regarding what?

File: docs/adr/013-add-provisioning-saga.md (hunk 1 above)
Comment (lzap) [reply docs/adr/013-add-provisioning-saga.md:17]: Threre is still `prop3os3ed` which is I believe not a word :-)

File: docs/adr/013-add-provisioning-saga.md (hunk 5 above)
Comment (ezr-ondrej) [reply docs/adr/013-add-provisioning-saga.md:59]: Where does the polling runs?

Comment (adiabramovitch) [reply docs/adr/013-add-provisioning-saga.md:59]: We are querying IB endpoint from the UI

Comment (ezr-ondrej) [reply docs/adr/013-add-provisioning-saga.md:59]: Here I was not making proposals (I don't love polling from UI, as closing a browser stops polling, which is not great).
Here I was asking if we could be clear on that in the text, so future readers know what were the proposals.

**Comment by adiabramovitch** [comment]
> Member

The explantation id different, I will change the title as well so it will be clearer

**Review by lzap — APPROVED** [review]

**Review by ezr-ondrej — APPROVED** [review]
Thanks @adiabramovitch lets merge this and finish it off once we have decision on the pending state :+1:

**Pull request merged**